language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/changetype/ChangeVariableTypeCodeAction.java

```diff
@@ -151,6 +151,10 @@ private Optional<NonTerminalNode> getVariableOrObjectFieldNode(NonTerminalNode s
             return Optional.of(sNode);
         } else if (isVariableNode(sNode.parent()) || sNode.parent().kind() == SyntaxKind.OBJECT_FIELD) {
             return Optional.of(sNode.parent());
+        } else if (sNode.kind() == SyntaxKind.LIST_CONSTRUCTOR && sNode.parent().kind() == SyntaxKind.COLLECT_CLAUSE
```

Comment:
why do we access the parent like this? Is this capture all the possible scenarios?

Reply:
Updated the logic to be more generic for a collect clause.